From 3c73ec63262ac7ca93c52c8fdad8b6d5318b6eec Mon Sep 17 00:00:00 2001
From: Alona Solntseva <alona.solntseva@tandemg.com>
Date: Mon, 21 Mar 2022 17:22:43 +0200
Subject: [PATCH] [SW-52631] gaudi2: revert legacy version registers initialization

Those registers are used in erom to obtain preboot version
Fuse version is transferred over the COMMS protocol, and the 
legacy register definition was removed For backwards compatibility
fuse_ver_offset and preboot_ver_off are still initialized
here to the legacy scratchpad register

Change-Id: Ieff30b94f866d02818195cbe44f0bbb8298cd044
Signed-off-by: Alona Solntseva <alona.solntseva@tandemg.com>
---

diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index c710df4..52747a6 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -1300,8 +1300,14 @@
 	global_conf.upd_pending_sts_map = mmUPD_PENDING_STS;
 	global_conf.upd_cmd_map = mmUPD_CMD;
 	global_conf.upd_sts_map = mmUPD_STS;
-	global_conf.preboot_ver_off_map = 0;	/* unused */
-	global_conf.mmfuse_ver_off_map = 0;	/* unused */
+	/*
+	 * Fuse version is transferred over the COMMS protocol, and the legacy
+	 * register definition was removed. For backwards compatibility
+	 * fuse_ver_offset and preboot_ver_off are still initialized here to
+	 * the legacy scratchpad register.
+	 */
+	global_conf.preboot_ver_off_map = mmPSOC_GLOBAL_CONF_SCRATCHPAD_28;
+	global_conf.mmfuse_ver_off_map = mmPSOC_GLOBAL_CONF_SCRATCHPAD_22;
 }
 
 void device_init_ca53_cfg(void)
